spynnaker/pyNN/models/neural_projections/projection_machine_edge.py

```diff
@@ -56,8 +56,38 @@ def filter_edge(self, graph_mapper):
                 pre_hi = graph_mapper.get_slice(self.pre_vertex).hi_atom
                 post_lo = graph_mapper.get_slice(self.post_vertex).lo_atom
                 post_hi = graph_mapper.get_slice(self.post_vertex).hi_atom
-                if pre_hi < post_lo or pre_lo > post_hi:
-                    n_filtered += 1
+                # Filter edge if both are views and outside limits
+                if ((synapse_info.prepop_is_view) and
+                        (synapse_info.postpop_is_view)):
+                    prepop_lo = synapse_info.pre_population._indexes[0]
+                    prepop_hi = synapse_info.pre_population._indexes[-1]
+                    postpop_lo = synapse_info.post_population._indexes[0]
+                    postpop_hi = synapse_info.post_population._indexes[-1]
+                    if ((prepop_hi < pre_lo or prepop_lo > pre_hi) or
+                            (postpop_hi < post_lo or postpop_lo > post_hi)):
```

Comment:
Not sure this is right - this seems to check that the population view is in range of the population.  Shouldn't it check pre_hi/lo against post_hi/lo?

Reply:
This is checking whether the values from the indices of a PopulationView are outside the current slice which the View could be a part of.  So it needs to check prepop_view_hi/lo vs. pre_slice_lo/hi (and likewise for post).

Reply:
OK, yes, I think I see this now.  Although not wrong, would this filter out everything?  For example, if I have two populations of 6 neurons with 2 neurons per core, and connect a[0:4] to b[2:6], as follows:
```
a: |01|23|--|
b: |--|01|23|
```
There will now be an edge between machine vertices a[2:3] and b[2:3].  Both of these have something in the pre and post slices that are in range, but there will be no synapses on this edge I think.

A possible thought I had was to subtract ```prepop_lo``` from ```pre_lo``` and ```pre_hi``` and correspondingly ```postpop_lo``` from ```post_lo``` and ```post_hi```.  You can then compare the results of these to check for the cross over.  I think you still need the above check as well though.

Reply:
I think I've got this working correctly now.